server/api/response.go

```diff
@@ -52,9 +54,24 @@ func (rsp *Response) Write(b []byte) (int, error) {
 	return n, nil
 }
 
-func RedirectResponse(url string) Response {
+// Returns a redirect response to a URL, optionally with a query.
+// Multiple queries will be merged, duplicate keys will be merged.
+func RedirectResponse(to string, qs ...url.Values) Response {
+	if len(qs) > 0 {
+		q := qs[0]
+		for i := 1; i < len(qs); i++ {
+			for k, v := range qs[i] {
+				q[k] = append(q[k], v...)
+			}
+		}
+		separator := "?"
+		if strings.Contains(to, separator) {
+			separator = "&"
+		}
+		to += separator + q.Encode()
```

Comment:
Pedantic thought: What happens if the redirect URL contains a fragment?

This seems like a really minor thing to worry about though

Reply:
```
u, err := url.Parse(to)
if err != nil {
    ...
}

if len(u.RawQuery) > 0 {
    u.RawQuery += "&"
}
u.RawQuery += q.Encode()

to = url.String()
```
maybe?

Reply:
(kicking bors)